### ISSUE TITLE
Error with dirname and LOAD_PATH

### ISSUE THREAD

**Issue by sbacelar** [nicolaspayette/NetLogo.jl#2]
When I run the first line of code (I am using `vscode`editor):
`dirname(@__FILE__) ∉ LOAD_PATH && push!(LOAD_PATH, dirname(@__FILE__))`

I get the following error:
```
ERROR: MethodError: no method matching dirname(::Void)
Closest candidates are:
  dirname(::AbstractString) at path.jl:146
Stacktrace:
 [1] eval(::Module, ::Any) at ./boot.jl:235
```

**Comment by nicolaspayette** [comment]
Try running `Pkg.update()`. You probably installed the package before I removed that line in `demo.jl`. Let me know if it works...

**Comment by nicolaspayette** [comment]
(And make sure you get `demo.jl` from your `.julia/v0.6/NetLogo/` directory, instead of using a version you might have downloaded previously.)

**Comment by sbacelar** [comment]
I am running demo.jl from .julia/v0.6/Netlogo/ but I get the following errors when `using NetLogo`:
```
using NetLogo
Main> using NetLogo
ERROR: ArgumentError: Module NetLogo not found in current path.
Run `Pkg.add("NetLogo")` to install the NetLogo package.
Stacktrace:
 [1] _require(::Symbol) at ./loading.jl:435
 [2] require(::Symbol) at ./loading.jl:405
 [3] eval(::Module, ::Any) at ./boot.jl:235

Main> Pkg.add("NetLogo")
INFO: Package NetLogo is already installed

Main> using NetLogo
ERROR: ArgumentError: Module NetLogo not found in current path.
Run `Pkg.add("NetLogo")` to install the NetLogo package.
Stacktrace:
 [1] _require(::Symbol) at ./loading.jl:435
 [2] require(::Symbol) at ./loading.jl:405
 [3] eval(::Module, ::Any) at ./boot.jl:235
```

**Comment by nicolaspayette** [comment]
Can you try removing it and then recloning?:

```julia
Pkg.rm("NetLogo")
Pkg.clone("git@github.com:nicolaspayette/NetLogo.jl.git")
```

And then maybe restart Julia.

Sorry if there is a bit of messing around. I am not yet very familiar with how to setup a package so other people can use it. It's quite possible that I made a mistake somewhere along the way. I appreciate you helping me getting this up and running...

**Comment by sbacelar** [comment]
It seems it doesn't work yet:

```
Main> Pkg.rm("NetLogo")
INFO: Removing NetLogo (unregistered)

Main> Pkg.clone("git@github.com:nicolaspayette/NetLogo.jl.git")
INFO: Cloning NetLogo from git@github.com:nicolaspayette/NetLogo.jl.git
Passphrase for /Users/sergiobacelar/.ssh/id_rsa:
INFO: Computing changes...
INFO: No packages to install, update or remove

Main> Pkg.add("NetLogo")
INFO: Package NetLogo is already installed
INFO: METADATA is out-of-date — you may not have the latest version of NetLogo
INFO: Use `Pkg.update()` to get the latest versions of your packages

Main> Pkg.update()
INFO: Updating METADATA...
INFO: Updating Homebrew master...
Passphrase for /Users/sergiobacelar/.ssh/id_rsa:
INFO: Updating NetLogo master...
INFO: Computing changes...
INFO: No packages to install, update or remove

Main> using NetLogo
ERROR: ArgumentError: Module NetLogo not found in current path.
Run `Pkg.add("NetLogo")` to install the NetLogo package.
Stacktrace:
 [1] _require(::Symbol) at ./loading.jl:435
 [2] require(::Symbol) at ./loading.jl:405
 [3] eval(::Module, ::Any) at ./boot.jl:235
```

**Comment by nicolaspayette** [comment]
OK, I think I needed to move `NetLogo.jl` to the `src` folder. Can you run `Pkg.update()` and try again?

**Comment by sbacelar** [comment]
Now it works. I had to install JRE legacy code on my Mac. But I had to change one line of your code (`NetLogo.init`) and I got an error. Please see what is the mistake I did:
```
Main> using NetLogo
Loaded /Library/Java/JavaVirtualMachines/jdk1.8.0_144.jdk/Contents/Home/jre/lib/server/libjvm.dylib

Main> NetLogo.init("/Applications/Netlogo 6.0.4/Netlogo 6.0.4.app")

signal (11): Segmentation fault: 11
while loading no file, in expression starting on line 0
unknown function (ip: 0x11dad72b3)
Allocations: 4188932 (Pool: 4187476; Big: 1456); GC: 7
```

**Comment by nicolaspayette** [comment]
Thanks for you patience and your perseverance! I'm glad the installation path problem is solved. Would you mind opening a new separate issue for the segfault problem? I'm going to close this one, which I consider to be fixed by 1b7c7d7b59aa10c5707f024c7f86ebe0c257c228.
